paddle/fluid/framework/grad_op_desc_maker.h

```diff
@@ -132,22 +141,47 @@ class GradOpDescMakerBase {
   std::string ForwardOpType() const { return this->fwd_op_.Type(); }
 
  protected:
-  const OpDesc& ForwardOp() const { return fwd_op_; }
+  // const OpDesc& ForwardOp() const { return fwd_op_; }
+
+  bool HaveInput(const std::string& name) const {
```

Comment:
`HaveInput` -> `HasInput`. 

BTW, as you have defined `GradOpBaseMakerBase`, we do not need any `var_base_in_` and `var_base_out_`.